src/background/browser-permissions-tracker.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { Interpreter } from 'background/interpreter';
+import { BrowserAdapter } from 'common/browser-adapters/browser-adapter';
+import { Logger } from 'common/logging/logger';
+import { Message } from 'common/message';
+import { Messages } from 'common/messages';
+
+export const allUrlAndFilePermissions: string = '*://*/*';
+
+export class BrowserPermissionsTracker {
+    constructor(
+        private browserAdapter: BrowserAdapter,
+        private interpreter: Interpreter,
+        private readonly logger: Logger,
+    ) {}
+
+    public async initialize(): Promise<void> {
+        this.browserAdapter.addListenerOnPermissionsAdded(this.notifyChange);
+        this.browserAdapter.addListenerOnPermissionsRemoved(this.notifyChange);
+        await this.notifyChange();
+    }
+
+    private notifyChange = async (): Promise<void> => {
```

Comment:
notifyChange is probably not a great name for this. Something like "updatePermissionState" or something might be more appropriate.